Classes/Service/NodePolicyService.php

```diff
@@ -114,14 +115,68 @@ public function getDisallowedNodeTypes(NodeInterface $node): array
             return $disallowedNodeTypes;
         }
 
+        // determine the set of configured node supertypes
+        $nodeNodeType = $node->getNodeType();
+
+        $superTypes = [];
+        $generateSuperTypes = function (array $nodeTypes, &$superTypes) use (&$generateSuperTypes) {
+            foreach ($nodeTypes as $nodeType) {
+                $superTypes[$nodeType->getName()] = $nodeType;
+                $generateSuperTypes($nodeType->getDeclaredSuperTypes(), $superTypes);
+            }
+        };
+        $generateSuperTypes($nodeNodeType->getDeclaredSuperTypes(), $superTypes);
+
+        $parentNodeType = null;
+        try {
```

Comment:
Couldn't this try/catch be replaced by a `getParent` and checking whether its not null?

Reply:
Yes that would be possible, but `getParent()` is deprecated and in the comment for `getParent()` it is mentioned that you should use `findParentNode()`.

Reply:
True, just wondered if the try/catch stuff might have a negative impact again. But not sure how many times the method is actually called.

Reply:
https://php.watch/articles/php-exception-performance

just less performant, when an exception will be thrown.

Reply:
but an exception will only be thrown when `$this->isRoot() === true`:
https://github.com/neos/neos-development-collection/blob/c14cc4ab46e7d9be75157bb5aa80f96fcc2dff14/Neos.ContentRepository/Classes/Domain/Model/Node.php#L2026

so we could instead check for that first and never get an exception.